### ISSUE TITLE
Fix S3247 FN: Support list patterns 

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6287]
### Description

FN for S3247 (CastShouldNotBeDuplicated).

Using C#11 list patterns, the rule is not raising an issue.

### Repro steps

```csharp
object[] numbers = { 1, 2, 3 };
if (numbers is [Fruit fruit, 3, 3])    // FN
{
    var ff1 = (Fruit)fruit;            // FN
}
```

### Expected behavior

The rule should raise an issue, since here we are casting **twice** to Fruit.

### Actual behavior

The rule does not raise an issue.
